src/mongoc/mongoc-server-description-private.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2014 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MONGOC_SERVER_DESCRIPTION_PRIVATE_H
+#define MONGOC_SERVER_DESCRIPTION_PRIVATE_H
+
+#include "mongoc-server-description.h"
+
+typedef enum
+   {
+      MONGOC_SERVER_UNKNOWN,
+      MONGOC_SERVER_STANDALONE,
+      MONGOC_SERVER_MONGOS,
+      MONGOC_SERVER_POSSIBLE_PRIMARY,
+      MONGOC_SERVER_RS_PRIMARY,
+      MONGOC_SERVER_RS_SECONDARY,
+      MONGOC_SERVER_RS_ARBITER,
+      MONGOC_SERVER_RS_OTHER,
+      MONGOC_SERVER_RS_GHOST,
+      MONGOC_SERVER_DESCRIPTION_TYPES,
+   } mongoc_server_description_type_t;
+
+struct _mongoc_server_description_t
+{
+   uint32_t                         id;
+   mongoc_host_list_t               host;
+   int64_t                          round_trip_time;
+   bson_t                           last_is_master;
+
+   /* The following fields are filled from the last_is_master and are zeroed on
+    * parse.  So order matters here.  DON'T move set_name */
+   const char                      *set_name;
+   const char                      *connection_address;
+   char                            *error; // TODO: what type should this be?
+   mongoc_server_description_type_t type;
+   int32_t                          min_wire_version;
+   int32_t                          max_wire_version;
```

Comment:
Let's talk about whether it's safe to use monitoring to discover wire version, max_write_batch_size, etc.: there were races in PyMongo that I solved by *not* using monitoring to find these values, and instead calling "ismaster" on each new socket during socket setup and using those per-socket values for formatting messages to send on that socket.